rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -201,7 +240,8 @@ public void getRecoveringLRAs(Map<URI, LongRunningAction> lras) {
      */
     public boolean removeCommitted(Uid lraUid) {
         try {
-            return _recoveryStore.remove_committed(lraUid, _transactionType);
+            return _recoveryStore.remove_committed(lraUid, _transactionType) ||
+                    _recoveryStore.remove_committed(lraUid, LongRunningAction.getFailedLraType());
```

Comment:
Why are we removing failed records?

Reply:
Because `removeCommitted` is called from LRAService#removeLog (https://github.com/jbosstm/narayana/blob/master/rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java#L188)
and `LRAService#removeLog` is called from RecoveryCoordinator#deleteFailedLRA (https://github.com/jbosstm/narayana/blob/master/rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/RecoveryCoordinator.java#L205)

Here we are checking both the location as explained in previous comments.

Reply:
You should only be removing the original record. Do not delete failure records from the new location.

Reply:
okay